### PR TITLE
add Timer

### PR DESCRIPTION
**Pull request by MateuszBlach**
shared timer that can be used in GPU/CPU algorithm to measure the time

### CODE REVIEW

File: shared/timer/Timer.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "Timer.h"
```
Comment (Jocho14) [inline shared/timer/Timer.cpp:1]: adjust import

Comment (Jocho14) [inline shared/timer/Timer.cpp]: change to .hpp file

File: shared/timer/Timer.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+double Timer::getTime() const {
+    if (running) {
+        auto currentTime = std::chrono::high_resolution_clock::now();
+        return std::chrono::duration<double, std::milli>(currentTime - startTime).count();
+    } else {
+        return std::chrono::duration<double, std::milli>(endTime - startTime).count();
+    }
+}
+
+void Timer::resetTime() {
```
Comment (Jocho14) [inline shared/timer/Timer.cpp]: rename to Timer::reset()
```suggestion
void Timer::reset() {
```

File: shared/timer/Timer.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#ifndef TIMER_H
```
Comment (Jocho14) [inline shared/timer/Timer.h:1]:
```suggestion
#ifndef TIMER_HPP
```

Comment (Jocho14) [inline shared/timer/Timer.h]: rename to TIMER_HPP

File: shared/timer/Timer.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+#define TIMER_H
```
Comment (Jocho14) [inline shared/timer/Timer.h]: rename to TIMER_HPP

File: shared/timer/Timer.h [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    double getTime() const;
+    void resetTime();
+
+private:
+    std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
+    std::chrono::time_point<std::chrono::high_resolution_clock> endTime;
+    bool running;
+};
+
+#endif
```
Comment (Jocho14) [inline shared/timer/Timer.h]: Add corresponding ifndef as a comment - #endif  // TIMER_HPP

Comment (Jocho14) [inline shared/timer/Timer.h]: leave a blank line at the end of the file

File: shared/timer/Timer.cpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
+#include "Timer.h"
+
+Timer::Timer() : running(false) {}
```
Comment (Jocho14) [inline shared/timer/Timer.cpp]: initialize startTime and endTime using default initializer:
startTime(), endTime()

**Review by Jocho14 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by MateuszBlach** [comment]
check now @Jocho14

File: shared/timer/Timer.hpp [hunk 7]
```diff
@@ -10,12 +10,13 @@ class Timer {
     void start();
     void stop();
     double getTime() const;
-    void resetTime();
+    void reset();
 
 private:
     std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
```
Comment (Jocho14) [inline shared/timer/Timer.hpp]: add underscore suffix "_" in every private member variable. For example:
`std::chrono::time_point<std::chrono::high_resolution_clock> start_time_;`

**Review by Jocho14 — CHANGES_REQUESTED** [review]
Please resolve the rest of the comments

**Comment by MateuszBlach** [comment]
check now @Jocho14

**Comment by MateuszBlach** [comment]
@Jocho14 check now

**Comment by Jocho14** [comment]
+2

**Pull request merged**